packages/core/src/guards.ts

```diff
@@ -104,14 +92,19 @@ export function evaluateGuard<
   guard: GuardDefinition<TContext, TEvent>,
   context: TContext,
   _event: SCXML.Event<TEvent>,
-  state: State<TContext, TEvent>
+  state: State<TContext, TEvent>,
+  actorContext?: ActorContext<any, any>
 ): boolean {
   const { machine } = state;
   const guardMeta: GuardMeta<TContext, TEvent> = {
     state,
     guard,
     _event,
-    evaluate: evaluateGuard
+    evaluate: evaluateGuard,
+    // TODO: `self` should always be "defined", and developers should expect it to exist
+    // only when they're actually interpreting the machine.
+    // Trying to access `self` with an uninterpreted machine should fail correctly.
```

Comment:
it's nice to have the exact same shape of "meta" stuff in all implementation types. I wonder though - what are practical use cases for having access to `self` in guards? Remember that having it here impacts purity/idempotency and that has negative consequences for APIs like `state.can`